transformations/butter_fingers_perturbation_for_IL/transformation.py

```diff
@@ -0,0 +1,105 @@
+import os
+import itertools
+import random
+import json
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+'''
+    This transformation adds noise to all types of text sources (sentence, paragraph, etc.), which is proportional to noise erupting from keyboard typos resulting in common spelling errors. 
+    We have expanded the existing implementation for English (source) to a few Indian Languages:
+    1. Bangla ('bn'), 
+    2. Gujarati ('gu'), 
+    3. Hindi ('hi'), 
+    4. Kannada ('kn'), 
+    5. Malayalam ('ml'), 
+    6. Oriya ('or'), 
+    7. Punjabi ('pa'), 
+    8. Tamil ('ta'), 
+    9. Telugu ('te')
+
+    args:
+        language (str): base language for butter fingers
+        keyboard (str): preffered keyboard schema (currently available: "inscript")
+        seed (int, default = 0): seed for reproducibility 
+'''
+
+def butter_finger(text, language:str, keyboard:str, prob=0.1,  seed=0, max_outputs=1):
+
+    mapping_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'key_mapping.json')
+
+    with open(mapping_path,'r') as f:
+        key_mapping = json.load(f)
+
+    languages = list(key_mapping.keys())
+
+    if language not in key_mapping:
+        raise Exception(f"Language not supported. Available languages: {languages}.")
+
+    keyboards = list(key_mapping[language].keys())
+    if keyboard not in key_mapping[language]:
+        raise Exception(f"Keyboard not supported. Available keyboards for '{language}': {keyboards}.")
+
+    random.seed(seed)
+    key_approx = key_mapping[language][keyboard]
```

Comment:
This section can be moved to the class initialization, that way once you have instantiated the transformation you can all it multiple times in different texts without having to run this portion of code every time.

Reply:
Have now shifted this code block into the class init and edited the generate and butter_fingers functions accordingly.